pkg/server/routes.go

```diff
@@ -30,20 +32,20 @@ func setupRoutes(cfg *config.Config, authChecker auth.Checker) *mux.Router {
 		})
 	})
 	api.HandleFunc("/status", handler.Status)
-	api.HandleFunc("/loki/ready", handler.LokiReady(&cfg.Loki))
-	api.HandleFunc("/loki/metrics", handler.LokiMetrics(&cfg.Loki))
-	api.HandleFunc("/loki/buildinfo", handler.LokiBuildInfos(&cfg.Loki))
-	api.HandleFunc("/loki/config/limits", handler.LokiConfig(&cfg.Loki, "limits_config"))
-	api.HandleFunc("/loki/config/ingester/max_chunk_age", handler.IngesterMaxChunkAge(&cfg.Loki))
-	api.HandleFunc("/loki/flow/records", handler.GetFlows(cfg))
-	api.HandleFunc("/loki/flow/metrics", handler.GetTopology(cfg))
-	api.HandleFunc("/loki/export", handler.ExportFlows(cfg))
-	api.HandleFunc("/resources/clusters", handler.GetClusters(&cfg.Loki))
-	api.HandleFunc("/resources/zones", handler.GetZones(&cfg.Loki))
-	api.HandleFunc("/resources/namespaces", handler.GetNamespaces(&cfg.Loki))
-	api.HandleFunc("/resources/namespace/{namespace}/kind/{kind}/names", handler.GetNames(&cfg.Loki))
-	api.HandleFunc("/resources/kind/{kind}/names", handler.GetNames(&cfg.Loki))
-	api.HandleFunc("/frontend-config", handler.GetFrontendConfig(cfg.Frontend.BuildVersion, cfg.Frontend.BuildDate, cfg.Path))
+	api.HandleFunc("/loki/ready", h.LokiReady())
```

Comment:
would it make sense to check here
```go
if cfg.IsLokiEnabled() {
        api.HandleFunc("/loki/ready", h.LokiReady())
        api.HandleFunc("/loki/metrics", h.LokiMetrics())
	api.HandleFunc("/loki/buildinfo", h.LokiBuildInfos())
	api.HandleFunc("/loki/config/limits", h.LokiConfig("limits_config"))
	api.HandleFunc("/loki/config/ingester/max_chunk_age", h.IngesterMaxChunkAge())
	api.HandleFunc("/loki/flow/records", h.GetFlows(ctx))
	api.HandleFunc("/loki/flow/metrics", h.GetTopology(ctx))
	api.HandleFunc("/loki/export", h.ExportFlows(ctx))
}
```

and with that u can remove the same check done in each of those function ?

Reply:
I guess it's possible, but on the other hand, I find it more informative than a 404 error to tell explicitly that the endpoint is not available due to Loki being disabled in an error message